### PR TITLE
fix aliasing on unwrap optional

### PR DESCRIPTION
**Pull request by eellison**
Fix for https://github.com/pytorch/pytorch/issues/15604

### CODE REVIEW

**Review by driazati — APPROVED** [review]

**Review by suo — APPROVED** [review]
thanks

File: torch/csrc/jit/passes/alias_analysis.cpp [hunk 1]
```diff
@@ -366,6 +367,11 @@ void AliasDb::analyze(Node* node) {
       continue;
     }
 
+    // If this type cannot alias, continue. Can occur with a VarType schema
+    if (!shouldAnnotate(actualValue)) {
```
Comment (suo) [inline torch/csrc/jit/passes/alias_analysis.cpp]: what problem are you trying to solve by adding these? Looks like if `actualValue` is a vartype, it'll pass anyway

Comment (eellison) [reply torch/csrc/jit/passes/alias_analysis.cpp]: At this point the vartype is bound to an actual type. I'm not sure if we need to add VarType within shouldAnnotate, but it seems safer to do so.

Comment (suo) [reply torch/csrc/jit/passes/alias_analysis.cpp]: sounds good to me

**Comment by suo** [comment]
@eellison "retest this please" only works for travis tests, and since they test against the PR base revision on not master now it probably doesn't do what you want anymore?

**Comment by eellison** [comment]
> @eellison "retest this please" only works for jenkins tests, and since they test against the PR base revision on not master now it probably doesn't do what you want anymore?

It looks like maybe it's a CI Failure 

```/usr/local/Homebrew/Library/Homebrew/config.rb:39:in `initialize': no implicit conversion of nil into String (TypeError)
	from /usr/local/Homebrew/Library/Homebrew/config.rb:39:in `new'
	from /usr/local/Homebrew/Library/Homebrew/config.rb:39:in `<top (required)>' 
```

**Comment by suo** [comment]
rebase past this: https://github.com/pytorch/pytorch/commit/fe15d6a2c231a7bc1b32781217ed336ccf9adff7

**Pull request closed without merging**